service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/AllergyStructureMapper.java

```diff
@@ -193,6 +195,16 @@ private String buildAsserterPertinentInformation(AllergyIntolerance allergyIntol
             IIdType reference = allergyIntolerance.getAsserter().getReferenceElement();
             if (reference.getResourceType().equals(ResourceType.Patient.name())) {
                 return PATIENT_ASSERTER;
+            } else if (reference.getResourceType().equals(ResourceType.RelatedPerson.name())) {
+                RelatedPerson relatedPerson = messageContext
+                    .getInputBundleHolder()
+                    .getResource(reference)
+                    .map(RelatedPerson.class::cast)
+                    .get();
```

Comment:
What if there is no such resource?
It should be something like this:
```
return messageContext
    .getInputBundleHolder()
    .getResource(reference)
    .map(RelatedPerson.class::cast)
    .filter(RelatedPerson::hasName)
    .map(RelatedPerson::getName)
    .map(names -> names.get(0))
    .map(HumanName::GetNameAsSingleString)
    .map(name -> String.format(RELATED_PERSON_ASSERTER, name))
    .orElse(StringUtils.EMPTY);
```

Reply:
Addressed